crates/context_aware_config/src/api/config/handlers.rs

```diff
@@ -527,7 +529,8 @@ async fn reduce_config(
         .and_then(|value| value.to_str().ok().and_then(|s| s.parse::<bool>().ok()))
         .unwrap_or(false);
 
-    let dimensions_schema_map = get_all_dimension_schema_map(&mut conn)?;
+    let dimensions_vec = get_dimension_data(&mut conn)?;
```

Comment:
Should `get_dimension_data` return a map?

Reply:
No , it does not need to be , i have separated it into 2 function and get_dimension_data_map this function returns only values which are needed

Reply:
i have separated data fetch from db and business logic on db separate
lets we need to do multiple operation on dimension list in a single handler , then we wont have to do db call everytime

Reply:
but then do we want to start separating db calls to separate functions yet  ?

Reply:
no harm, just wanted clarification here